Makefile

```diff
@@ -90,18 +91,35 @@ docker-push: ## Push docker image with the manager.
 	docker push ${IMG}
 
 ##@ Deployment
+PLUGIN=
+ensure-plugin:
+ifeq ("$(PLUGIN)", "")
+	$(error "Select the plugin using PLUGIN make variable.")
+else ifeq (,$(wildcard ./config/plugins/$(PLUGIN)))
+	$(error "Unknown plugin $(PLUGIN)")
+endif
 
-deploy:  kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG} && $(KUSTOMIZE) edit set image kmra-plugin=${IMG}
-	$(KUSTOMIZE) build config/default | kubectl apply -f -
+## Deploy controller to the K8s cluster specified in ~/.kube/config.
+## If provided OUTFILE instead it writes the deployment to the given
+## file.
+deploy: kustomize ensure-plugin
+	$(eval TMP := $(shell mktemp -d -p ./config/))
+	echo -e "bases:\n- ../plugins/$(PLUGIN)" > $(TMP)/kustomization.yaml
+	cd $(TMP) && $(KUSTOMIZE) edit set image controller=${IMG} $(PLUGIN)-plugin=${IMG}
+ifneq ("$(OUTFILE)", "")
+	$(KUSTOMIZE) build $(TMP) -o $(OUTFILE)
+else
+	$(KUSTOMIZE) build $(TMP) | kubectl apply -f -
+endif
+	rm -rf $(TMP)
```

Comment:
Hmm, are you no longer in the $(TMP) directory at this point?

Reply:
No,  We are in $(TMP) folder only while `kustomize edit`. So, it is okay to remove it here.